sklearn/kernel_approximation.py

```diff
@@ -991,6 +991,11 @@ def fit(self, X, y=None):
         self : object
             Returns the instance itself.
         """
+        xp, _ = get_namespace(X)
+        if X.ndim < 2:
```

Comment:
We should not access `X.ndim` before the call to `self._validate_data(X, accept_sparse="csr")` a few lines below.

Furthermore, `self._validate_data` itself is in charge of checking the shape of `X` by passing appropriate values to `**check_params` if needed.

However, all scikit-learn estimators are expected to raise a standardized error message when input data is 1D. `self._validate_data` is in doing this automatically by default so there is no need to do it manually.